wolvic/browser/vr/wvr_manager.cc

```diff
@@ -403,15 +403,24 @@ WvrManager::GetInputSourceState() {
         controller.hand == mozilla::gfx::ControllerHand::Left
             ? device::mojom::XRHandedness::LEFT
             : device::mojom::XRHandedness::RIGHT;
-
     // TODO: Get from external
     input_source->description->profiles = {
         "oculus-touch-v3", "oculus-touch-v2", "oculus-touch",
         "generic-trigger-squeeze-thumbstick"};
 
-    gfx::Transform input_from_pointer;
-    WvrMatToTransform(controller.axisValue, &input_from_pointer);
-    input_source->description->input_from_pointer = input_from_pointer;
+    input_source->gamepad = device::Gamepad();
+    input_source->gamepad->buttons_length = controller.numButtons;
+    input_source->gamepad->hand =
+        controller.hand == mozilla::gfx::ControllerHand::Left
+            ? device::GamepadHand::kLeft
+            : device::GamepadHand::kRight;
+    for (uint32_t j = 0; j < controller.numButtons; ++j) {
```

Comment:
We should make sure we don't pass the kButtonsLengthCap (and show a warning if that happens).

Reply:
Done in ToGamepadHand()